### ISSUE TITLE
D&D Beyond: Dice Icon Duplicating 

### ISSUE THREAD

**Issue by TheSwanDragon** [kakaroto/Beyond20#228]
I've found a small bug with D&D Beyond where clicking on and off of a feature duplicates the automatic roll dice icon which shows up. 

tl;dr: Clicking between features duplicates dice icons, clicking on the dice icons gives inaccurate numbers of dice rolled and incorrectly names them in Roll20.

For example:
![image](https://user-images.githubusercontent.com/31599991/80651715-4390a080-8a2b-11ea-9e61-eaa47faa422c.png)

Clicking between these two features causes this to happen:

![image](https://user-images.githubusercontent.com/31599991/80651754-5905ca80-8a2b-11ea-9052-70cb84cceebf.png)
This was after clicking on **Healing Light**, then **Bonus Cantrips**, then **Healing Light** again.

If I click on the first three dice (in order) to roll them, it will roll 1d6 three times, saying "Healing Light," then 1d6 twice, then after that just once.

If I click on **Bonus Cantrips** first, the dice will _still_ show up, except clicking on the first die will roll a d6 while saying "Bonus Cantrips" the first time I click on it.
^That same issue occurs no matter what I click on first, for example if I click on the **Sneak Attack** block, then click on the die on **Healing Light**, it will roll a d6 saying "Sneak Attack."
^That same issue _also_ occurs even if I don't click on a different section first, and in fact will output whichever section I clicked on most recently before clicking on the die.

After a little bit more testing, it's actually being a lot less consistent than that, in terms of how many dice it rolls. I believe that it rolls based on how many dice icons are there (if I've clicked between features a total of 8 times, then click on the first dice icon, it rolls 8 dice. Then if I click on the second, it rolls 7 dice, and so on). The dice icon showing up happens every time, and it consistently gives weird and inaccurate dice rolls.

The extra dice icons can be removed by refreshing or changing which tab I'm looking at on the Character Sheet (e.g., switching from **Features and Traits** to **Equipment** and back.

If I _just_ click on **Healing Light** and then roll the single die icon that shows up, it rolls 1d6 normally, as the underline shows.

These dice also show up on all features that could roll a die (for example, the **Pact Magic** feature which describes my spell attack. Clicking on a different feature rolls the correct size die (e.g., **Pact Magic** rolls a d20 with the correct Spell Attack Mod). However, it consistently displays the name of the most recently clicked feature (so if I click on a die icon on my Arcane Trickster **Spellcasting** after clicking on the **Pact Magic** feature, it will roll with my AT Spell Attack Mod while saying "Pact Magic.")

Please let me know if you have any questions about what I wrote, or if you'd like more pictures. I'm sending these dice rolls into a Roll20 Game.

**Comment by Aeristoka** [comment]
Duplicate of #183

**Comment by TheSwanDragon** [comment]
> Duplicate of #183

I only searched through issues that were still open, my bad. Should I mark this as closed?

**Comment by Aeristoka** [comment]
> > Duplicate of #183
> 
> I only searched through issues that were still open, my bad. Should I mark this as closed?

Yep, that would probably best.

**Comment by kakaroto** [comment]
Well, it may have been a duplicate, but thanks for writing such a detailed report :)
Let me know if you encounter anything else!
